CONTRIBUTING.md

```diff
@@ -45,6 +45,107 @@ When documenting Python classes, we adhere to the convention of including docstr
+
+- Other `alibi_detect` objects should be cross-referenced using references of the form `` :role:`~object` ``, where 
+`role` is one of the roles listed in the [sphinx documentation](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-python-objects), 
+and `object` is the full path of the object to reference. For example, the `MMDDrift` detectors's `predict` method 
+would be referenced with `` :meth:`~alibi_detect.cd.mmd.MMDDrift.predict` ``. This will render as `MMDDrift.predict()` and 
+link to the relevant API docs page. The same convention can be used to reference objects from other libraries, providing the
+library is included in `intersphinx_mapping` in `doc/source/conf.py`. If the `~` is removed, the absolute object location will be
+rendered.
+
+- Variable values or examples of setting an argument to a specific values should be written in double back-ticks
+to facilitate readability as they are rendered in a block with orange font-color.
+   - ``` is set to ``True`` ```
+   - ``` A list of features for which to plot the ALE curves or ``'all'`` for all features. ```
+   - ``` The search is greedy if ``beam_size=1`` ```
+   - ``` if the result uses ``segment_labels=(1, 2, 3)`` and ``partial_index=1``, this will return ``[1, 2]``. ```
+   
+- Listing the possible values an argument can take.
+   - ``` Possible values are: ``'all'`` | ``'row'`` | ``None``. ```
+
+- Returning the name of the variable and its description - standard convention and renders well. Writing the 
+variable types should be avoided as it would be duplicated from variables typing.
+```
+Returns
+-------
+raw
+    Array of perturbed text instances.
+data
+    Matrix with 1s and 0s indicating whether a word in the text has not been perturbed for each sample.
+```
+
+- Returning only the description. When the name of the variable is not returned, sphinx wrongly interprets the 
+description as the variable name which will render the text in italic. If the text exceeds one line, ``` \ ``` need 
+to be included after each line to avoid introducing bullet points at the beginning of each row. Moreover, if for 
+example the name of a variable is included between single back-ticks, the italic font is canceled for all the words
+with the exception of the ones inbetween single back-ticks.
+```
+Returns
+-------
+If the user has specified grouping, then the input object is subsampled and an object of the same \
+type is returned. Otherwise, a `shap_utils.Data` object containing the result of a k-means algorithm \
+is wrapped in a `shap_utils.DenseData` object and returned. The samples are weighted according to the \
+frequency of the occurrence of the clusters in the original data.
+```
+ 
+- Returning an object which contains multiple attributes and each attribute is described individually. 
+In this case the attribute name is written between single back-ticks and the type, if provided, would be written in 
```

Comment:
Nitpick:

> and the type, **if provided**, would be written

Could we be a little more specific here? Not detailing types in the general case (since already done by `sphinx-autodoc-typehints`), but giving types when multiple objects are contained in another object (i.e. a `dict`), makes a lot of sense IMO. However, in the latter case, are we saying this should always be done? Or its optional?

Reply:
I think we're hinting that it's better to give types however it's not something we've done historically. I've [opened](https://github.com/SeldonIO/alibi-detect/issues/791) an issue to change this across the codebase and then perhaps we can change the language here to be a little stronger.